code/modules/hydroponics/hydroponics.dm

```diff
@@ -768,6 +798,8 @@
 	self_sustaining = !self_sustaining
 	idle_power_usage = self_sustaining ? 5000 : 0
 	to_chat(user, "<span class='notice'>You [self_sustaining ? "activate" : "deactivated"] [src]'s autogrow function[self_sustaining ? ", maintaining the tray's health while using high amounts of power" : ""].")
+	if(!self_sustaining)
+		RefreshParts()

```

Comment:
why is this here?

Also can you set it so it requires anchoring to switch self_sustain? when I try to move trays I have to use rightclick or else they turn on.

Reply:
Currently, turning on self-sustaining enables a slower nutrient drain rate at earlier part levels. By not refreshing parts when turned off however, you can enable and re-enable autogrow to give all trays a drain-rate on par with tier 2 parts. Throwing in a RefreshParts is a safety net here.

Reply:
shouldn't the safety net be like on process or something?

it looks like the bug would still exist if it switched via another aspect that isn't a user ctrl clicking it.